src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int count;
+    public static final int MAX_CAPACITY = 10;
+    private Object[] elements = new Object[MAX_CAPACITY];
     @Override
     public void add(T value) {
+        if (count >= elements.length) {
+            elements = grow();
+        }
+        elements[count] = value;
+        count++;
 
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexforAdd(index);
+        count++;
+        Object[] newArray = new Object[elements.length + 1];
+        if (count > MAX_CAPACITY) {
```

Comment:
The condition `count > MAX_CAPACITY` is incorrect for determining when to grow the array. It should be `count >= elements.length`.